07-mongodb-mongoose/02-rest-api/controllers/categories.js

```diff
@@ -0,0 +1,12 @@
+const Category = require('../models/Category');
+
+module.exports.categoryList = async function categoryList(ctx, next) {
+  try {
+    const categories = await Category.find({});
+
+    ctx.body = {categories};
+  } catch (e) {
+    ctx.status = 400;
+    ctx.body = 'Bad Request';
```

Comment:
Are there any patterns for error message? Any good practices? I don't feel like doing a gag

Reply:
The error message should be well balanced: it should give enough information to a client but do not expose any sensitive information (like stack trace).

From my previous experience I could recommend such error:
```
{
   code: 134, // some internal code 
   message: "Short message", //
   data: {}
}
```

Where:
`code` - internal error code, that should be documented and clients may do some assumption based on its
`data` - error-specific object to add more context for data. E.g. for the validation error, it might contain a list of properties that didn't pass the validation and a description of what exactly wrong with the property.